### PR TITLE
WIP: Challenge Help me Carry

### PR DESCRIPTION
**Pull request by reinzor**

### CODE REVIEW

File: challenge_help_me_carry/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Challenge Help Me Carry 2017
+Responsible: Tim Assman
+
+## Test Log
+
+
+## Setup
+amigo1/sergio1:
+
+    astart/sstart # (if not already running)
```
Comment (LoyVanBeek) [inline challenge_help_me_carry/README.md]: This is not longer the case, should be amigo-start

File: challenge_help_me_carry/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+amigo2/sergio1:
+
+    ahardware/shardware: base, spindle, arms (left-arm is used) head
+
+amigo1/sergio1:
+
+    amiddle/smiddle
```
Comment (LoyVanBeek) [inline challenge_help_me_carry/README.md]: According to the new launch structure, a challenge-specific launch file should be added to amigo_bringup.
Then, you can launch amigo-challenge-help-me-carry if you do it correctly :-)

File: challenge_help_me_carry/src/challenge_help_me_carry.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    def execute(self, userdata):
+
+        target_waypoint, target_radius = challenge_knowledge.commands.waypoint_for_room[userdata.target_room]
+
+        navigateToWaypoint = states.NavigateToWaypoint(self._robot, EntityByIdDesignator(self._robot, id=target_waypoint), target_radius)
+
+        return navigateToWaypoint.execute()
+
+
+def setup_statemachine(robot):
```
Comment (LoyVanBeek) [inline challenge_help_me_carry/src/challenge_help_me_carry.py]: Please take a look at some of the other challenges. The main entry point should also be a class inheriting from smach.Statemachine

Comment (MatthijsBurgh) [inline challenge_help_me_carry/src/challenge_help_me_carry.py]: This is the same as other challenges

**Comment by LoyVanBeek** [comment]
I branched of the scaffold and put up #209 to merge that bit into master already to have at least something on master.

File: challenge_help_me_carry/src/challenge_help_me_carry.py [hunk 4]
```diff
@@ -64,24 +80,44 @@ def __init__(self, robot):
...
+
+    def execute(self, userdata):
+        # TODO: reuse grab action to grab bag from operator
+        # states.ArmToJointConfig(robot, arm_designator, "carrying_pose"),
+        states.Say(self._robot, "Grabbing item")
+        userdata.target_room_out = userdata.target_room_in
+
+        return "success"
+
+class NavigateToRoom(smach.State):
```
Comment (MatthijsBurgh) [inline challenge_help_me_carry/src/challenge_help_me_carry.py]: I don't see the benefits of using this state instead of NavigateToWaypoint

**Comment by LoyVanBeek** [comment]
@Rayman in simulation, this now fails because of "self._robot.ed.learn _person(self._operator_name) method disappeared!, returning False"

People tell me you made it disappear or at least have a replacement in the works somewhere. What is that status on that?

**Pull request merged**